Bucketeer/Sources/Public/BKTEvaluation.swift

```diff
@@ -19,3 +21,64 @@ public struct BKTEvaluation: Equatable {
         case prerequisite = "PREREQUISITE"
     }
 }
+
+public struct BKTEvaluationDetails<T:Equatable>: Equatable {
+    public let featureId: String
+    public let featureVersion: Int
+    public let userId: String
+    public let variationId: String
+    public let variationName: String
+    public let variationValue: T
+    public let reason: Reason
+
+    public enum Reason: String, Codable, Hashable {
+        case target = "TARGET"
+        case rule = "RULE"
+        case `default` = "DEFAULT"
+        case client = "CLIENT"
+        case offVariation = "OFF_VARIATION"
+        case prerequisite = "PREREQUISITE"
+
+        public static func fromString(value: String) -> Reason {
```

Comment:
It might be good to have a test for the `fromString()`.

Reply:
I will add them

Reply:
updated in https://github.com/bucketeer-io/ios-client-sdk/pull/90/commits/d170ebe84a91bcc0b47b31cbfe310074d5e7cadd